### PR TITLE
Use the latest version of TChannel-java with Netty 4.1

### PR DESCRIPTION
**Pull request by junchaowu**
A small additional change comes with this PR is renaming BaseInboundHandler to AbstractInboundHandler to follow the naming convention.
@yarpc/java

### CODE REVIEW

File: yarpc-core/src/main/java/io/yarpc/transport/http/HTTPClientInboundHandler.java [hunk 1]
```diff
@@ -48,13 +48,13 @@ public void channelActive(ChannelHandlerContext ctx) throws Exception {
     }
 
     @Override
-    protected void messageReceived(ChannelHandlerContext channelHandlerContext, FullHttpResponse httpResponse) throws
+    protected void channelRead0(ChannelHandlerContext channelHandlerContext, FullHttpResponse httpResponse) throws
```
Comment (kriskowal) [inline yarpc-core/src/main/java/io/yarpc/transport/http/HTTPClientInboundHandler.java]: What does this name `channelRead0` mean? What’s the rationale for the 0.

File: yarpc-core/build.gradle [hunk 2]
```diff
@@ -4,21 +4,21 @@ apply plugin: 'com.github.kt3k.coveralls'
...
     compile group: 'org.apache.commons', name: 'commons-lang3', version: '3.4'
     compile group: 'joda-time', name: 'joda-time', version: '2.9.2'
-    compile group: 'com.uber.tchannel', name: 'tchannel-core', version: '0.3.4'
-    compile group: 'io.netty', name: 'netty-all', version: '5.0.0.Alpha2'
+    compile group: 'com.uber.tchannel', name: 'tchannel-core', version: '0.4.0'
+    compile group: 'com.google.guava', name: 'guava', version: '18.0'
     compile group: 'org.slf4j', name: 'log4j-over-slf4j', version: '1.7.16'
     compile group: 'org.slf4j', name: 'slf4j-api', version: '1.7.16'
-    compile group: 'ch.qos.logback', name: 'logback-classic', version: '1.1.5'
+    compile group: 'com.github.mfornos', name: 'humanize-slim', version: '1.2.2'
```
Comment (abhinav) [inline yarpc-core/build.gradle]: Is this actually being used?

**Comment by kriskowal** [comment]
Please make sure a Java practitioner reviews these. The channelRead0 name is dubious but otherwise, looks good to me.

File: yarpc-core/src/main/java/io/yarpc/transport/http/HTTPClientInboundHandler.java (hunk 1 above)
Comment (junchaowu) [inline yarpc-core/src/main/java/io/yarpc/transport/http/HTTPClientInboundHandler.java]: Netty 4.1 is different from Netty 5.0. The is the API breaking change they have. They have same functionality but different name.

File: yarpc-core/build.gradle (hunk 2 above)
Comment (junchaowu) [inline yarpc-core/build.gradle]: Those will be used in errors branch. I am just breaking that big diff into small ones now.

File: yarpc-core/src/main/java/io/yarpc/transport/http/HTTPClientInboundHandler.java (hunk 1 above)
Comment (junchaowu) [inline yarpc-core/src/main/java/io/yarpc/transport/http/HTTPClientInboundHandler.java]: http://netty.io/wiki/new-and-noteworthy-in-5.0.html#channelread0--messagereceived for more information

**Comment by abhinav** [comment]
The `channelRead0` name is not in our control. It's a poorly named interface in Netty 4. It has been renamed in 5.

**Comment by abhinav** [comment]
:ok:

File: yarpc-core/build.gradle (hunk 2 above)
Comment (abhinav) [inline yarpc-core/build.gradle]: Seems like an unnecessary dependency, but let's discuss that elsewhere. This diff is good to go.

**Pull request merged**